rules/framework.bzl

```diff
@@ -25,6 +25,7 @@ load("@build_bazel_rules_apple//apple/internal:apple_toolchains.bzl", "AppleMacT
 load("@build_bazel_rules_apple//apple/internal:swift_support.bzl", "swift_support")
 load("@build_bazel_rules_apple//apple/internal/utils:clang_rt_dylibs.bzl", "clang_rt_dylibs")
 load("@build_bazel_rules_apple//apple:providers.bzl", "AppleBundleInfo", "IosFrameworkBundleInfo")
+load("@build_bazel_rules_apple//apple/internal:providers.bzl", "new_applebundleinfo")
```

Comment:
Do we need to rely on yet another internal provider from rules_apple here? It seems like it would continue to work well on 5.x.x pretty easily w/o the change?

Reply:
Also the issue with this line is everyone will have to run a commit later than:
https://github.com/bazelbuild/rules_apple/pull/2194

Reply:
It's incompatible because the initializer intentionally fails here in `_make_banned_init`: ```AppleBundleInfo, new_applebundleinfo = provider(
    doc = ...,
    fields = {
        ...
    },
    init = _make_banned_init(provider_name = "AppleBundleInfo"),
)```

Keeping `rules_ios` close to the head of `rules_apple` was intentional with this change.

Reply:
Yeah what I'm wondering, is how to make it work backwards with older versions:

```
def _new_apple_ctx():
```

Though the initializer _moving to_ a private API which used to be public is another story 😮‍💨